src/main/java/com/direwolf20/buildinggadgets/client/gui/materiallist/MaterialListGUI.java

```diff
@@ -1,217 +1,253 @@
+
+    private int backgroundX;
+    private int backgroundY;
+    private ItemStack template;
+
+    private String title;
+    private int titleLeft;
+    private int titleTop;
+
+    private ScrollingMaterialList scrollingList;
+
+    private Button buttonClose;
+    private Button buttonSortingModes;
+    private Button buttonCopyList;
+
+    private int hoveringTextX;
+    private int hoveringTextY;
+    private List<String> hoveringText;
+
+    public MaterialListGUI(ItemStack template) {
+        super(MaterialListTranslation.TITLE.componentTranslation());
+        Preconditions.checkArgument(template.getItem() instanceof ITemplate);
+        this.template = template;
+    }
+
+    @Override
+    public void init() {
+        this.backgroundX = getXForAlignedCenter(0, width, BACKGROUND_WIDTH);
+        this.backgroundY = getYForAlignedCenter(0, height, BACKGROUND_HEIGHT);
+
+        this.title = MaterialListTranslation.TITLE.format();
+        this.titleTop = getYForAlignedCenter(backgroundY, getWindowTopY() + ScrollingMaterialList.TOP, font.FONT_HEIGHT);
+        this.titleLeft = getXForAlignedCenter(backgroundX, getWindowRightX(), font.getStringWidth(title));
+
+        this.scrollingList = new ScrollingMaterialList(this);
+        // Make it receive mouse scroll events, so that the player can use his mouse wheel at the start
+        this.setFocused(scrollingList);
+        this.children.add(scrollingList);
+
+        int buttonY = getWindowBottomY() - (ScrollingMaterialList.BOTTOM / 2 + BUTTON_HEIGHT / 2);
+        this.buttonClose = new Button(0, buttonY, 0, BUTTON_HEIGHT, MaterialListTranslation.BUTTON_CLOSE.format(), b -> Minecraft.getInstance().player.closeScreen());
+        this.buttonSortingModes = new Button(0, buttonY, 0, BUTTON_HEIGHT, scrollingList.getSortingMode().getLocalizedName(), (button) -> {
+            scrollingList.setSortingMode(scrollingList.getSortingMode().next());
+            buttonSortingModes.setMessage(scrollingList.getSortingMode().getLocalizedName());
+        });
+        this.buttonCopyList = new Button(0, buttonY, 0, BUTTON_HEIGHT, MaterialListTranslation.BUTTON_COPY.format(), (button) -> {
+            getMinecraft().keyboardListener.setClipboardString(stringify(Screen.hasControlDown()));
+            getMinecraft().player.sendStatusMessage(new TranslationTextComponent(MaterialListTranslation.MESSAGE_COPY_SUCCESS.getTranslationKey()), true);
+        });
+
+        // Buttons will be placed left to right in this order
+        this.addButton(buttonSortingModes);
+        this.addButton(buttonCopyList);
+        this.addButton(buttonClose);
+        this.children.add(scrollingList);
+        this.calculateButtonsWidthAndX();
+    }
+
+    private String stringify(boolean detailed) {
+        if (detailed)
+            return stringifyDetailed();
+        return stringifySimple();
+    }
+
+    private String stringifyDetailed() {
+        return scrollingList.children().stream()
```

Comment:
I intended to replace this with the TemplateHeader, for which the to json conversion hasn't been written yet.

Reply:
This for example has the advantage of seeing nbt data of the Items, which might be required to built Tiles...